### PR TITLE
CMake 3.16 Build Optimizations

### PR DESCRIPTION
**Pull request by Hoikas**
While working on the vcpkg buildsystem as described in #736, I discovered that our precompiled header trickery doesn't work with the Ninja generator as used by Visual Studio. Fortunately, CMake 3.16 added native support for precompiled headers, so `PLASMA_USE_PCH` has been updated to use the native CMake support if we are on CMake 3.16 or higher. This also means that we get PCH on gcc as well. Support for CMake 3.14 and 3.15 remains, but you no longer have precompiled headers on those old versions.

CMake 3.16 also added support for unity builds, which seems to greatly decrease the build time of several of our targets. While we could simply pass `CMAKE_UNITY_BUILD=TRUE` on the command line, it seems the codebase has several ODR violations. So, for now, I've gone with a per-target opt-in approach. Unity can be disabled with `-DPLASMA_UNITY_BUILD=OFF`.

### CODE REVIEW

File: Sources/Plasma/CoreLib/hsWindows.h [hunk 1]
```diff
@@ -84,6 +84,11 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 #       include <vld.h>
 #   endif // USE_VLD
 
+    // Conflicts with plDynamicTextMap
+#   ifdef DrawText
+#       undef DrawText
+#   endif
```
Comment (zrax) [inline Sources/Plasma/CoreLib/hsWindows.h:90]: It looks like you explicitly changed it to use `DrawTextA` though -- is this still needed?

File: Sources/Tools/MaxComponent/CMakeLists.txt [hunk 2]
```diff
@@ -107,7 +106,6 @@ set(MaxComponent_SOURCES
     plBehavioralComponents.cpp
     plBipedKiller.cpp
     plBlowComponent.cpp
-    plCameraComponent.cpp
```
Comment (zrax) [inline Sources/Tools/MaxComponent/CMakeLists.txt:110]: Not sure how the deletion of plCameraComponent is related to PCH/Unity builds...  Perhaps this should be a separate PR?

File: Sources/Tools/MaxComponent/plDistribComponent.cpp [hunk 3]
```diff
@@ -434,9 +433,9 @@ class plDistCompBitmapProc : public plRollupMgrProc
 
             cbox = GetDlgItem(hWnd, IDC_COMP_DISTRIB_PROBCOLORCHAN);
             int i;
-            for( i = 0; i < kNumColorChanOptions; i++ )
+            for( i = 0; i < std::size(kColorChanStrings); i++ )
```
Comment (zrax) [inline Sources/Tools/MaxComponent/plDistribComponent.cpp:436]: You're now just one step off from making a range for loop -- why stop here?  ;)

File: Sources/Tools/MaxComponent/plDistribComponent_old.cpp [hunk 4]
```diff
@@ -152,7 +150,7 @@ class plDistribComponentProc_old : public ParamMap2UserDlgProc
         case WM_INITDIALOG:
             cbox = GetDlgItem(hWnd, IDC_COMP_DISTRIB_PROBCOLORCHAN);
             int i;
-            for( i = 0; i < kNumColorChanOptions; i++ )
+            for( i = 0; i < std::size(kProbColorChanStrings); i++ )
```
Comment (zrax) [inline Sources/Tools/MaxComponent/plDistribComponent_old.cpp:153]: Same as above

File: Sources/Tools/MaxComponent/plCameraComponent.cpp [hunk 5]
```diff
@@ -1,180 +0,0 @@
...
-static plCameraCmdComponentProc gCameraCmdComponentProc;
-
-enum
-{
-    kCommandSetOffset,
-    kCommandSetFP,
-    kCommandSetFixedCam,
-};
-
-ParamBlockDesc2 gCameraCmdBlock
```
Comment (zrax) [inline Sources/Tools/MaxComponent/plCameraComponent.cpp:113]: This file does have a (presumably self-registering) global...  Are you sure it's actually unneeded?

File: Sources/Plasma/CoreLib/hsWindows.h (hunk 1 above)
Comment (Hoikas) [reply Sources/Plasma/CoreLib/hsWindows.h:90]: Yes, without this, `pyDynamicText::DrawText()` is impacted... But I could rename that method if you prefer.

File: Sources/Tools/MaxComponent/CMakeLists.txt (hunk 2 above)
Comment (Hoikas) [reply Sources/Tools/MaxComponent/CMakeLists.txt:110]: This file's contents were a duplicated subset of plCameraComponents.cpp, causing the unity build to fail. This was previously a warning per #689.

File: Sources/Tools/MaxComponent/plCameraComponent.cpp (hunk 5 above)
Comment (Hoikas) [reply Sources/Tools/MaxComponent/plCameraComponent.cpp:113]: See plCameraComponent**s**.cpp :wink:

File: Sources/Plasma/CoreLib/hsWindows.h (hunk 1 above)
Comment (zrax) [reply Sources/Plasma/CoreLib/hsWindows.h:90]: Ah, I missed what was conflicting here...  Nevermind then

File: Sources/Tools/MaxComponent/CMakeLists.txt (hunk 2 above)
Comment (zrax) [reply Sources/Tools/MaxComponent/CMakeLists.txt:110]: *sigh*

**Review by zrax — APPROVED** [review]
